.drone.star

```diff
@@ -313,3 +319,153 @@ def appBuilds(ctx):
+                },
+            ],
+        },
+    ]
+
+    wait_for_ocis = [
+        {
+            "name": "wait-for-ocis",
+            "image": OC_CI_ALPINE,
+            "commands": [
+                "timeout 200 bash -c 'while [ $(curl -sk -uadmin:admin " +
+                "%s/graph/v1.0/users/admin " % environment["OCIS_URL"] +
+                "-w %{http_code} -o /dev/null) != 200 ]; do sleep 1; done'",
+            ],
+        },
+    ]
+
+    return app_build_steps + ocis_service + wait_for_ocis
+
+def uploadTracingResult(ctx):
+    return [{
+        "name": "upload-tracing-result",
+        "image": PLUGINS_S3,
+        "pull": "if-not-exists",
+        "settings": {
+            "bucket": {
+                "from_secret": "cache_public_s3_bucket",
+            },
+            "endpoint": {
+                "from_secret": "cache_public_s3_server",
+            },
+            "path_style": True,
+            "source": "test-results/**/*",
+            "strip_prefix": "test-results",
+            "target": "/${DRONE_REPO}/${DRONE_BUILD_NUMBER}/tracing",
+        },
+        "environment": {
+            "AWS_ACCESS_KEY_ID": {
+                "from_secret": "cache_public_s3_access_key",
+            },
+            "AWS_SECRET_ACCESS_KEY": {
+                "from_secret": "cache_public_s3_secret_key",
+            },
+        },
+        "when": {
+            "status": ["failure"],
+        },
+    }]
+
+def logTracingResult(ctx):
+    return [{
+        "name": "log-tracing-result",
+        "image": OC_UBUNTU,
+        "commands": [
+            "cd test-results/",
+            'echo "To see the trace, please open the following link in the console"',
+            'for f in */; do echo "npx playwright show-trace https://cache.owncloud.com/public/${DRONE_REPO}/${DRONE_BUILD_NUMBER}/tracing/$f"trace.zip" \n"; done',
+        ],
+        "when": {
+            "status": ["failure"],
+        },
+    }]
+
+def e2eTests(ctx):
+    e2e_test_steps = []
+
+    for app in APPS_COVERED_E2E:
+        e2e_test_steps.append({
+            "name": app,
+            "image": OC_CI_NODEJS,
+            "commands": [
+                "pnpm exec playwright install chromium",
+                "BASE_URL_OCIS=https://ocis:9200 pnpm test:e2e --project='%s-chromium'" % app,
+            ],
+        })
+
+    return [{
+        "kind": "pipeline",
+        "type": "docker",
+        "name": "e2e-tests",
+        "steps": installPnpm() + ocisService() + e2e_test_steps + uploadTracingResult(ctx) + logTracingResult(ctx),
+        "trigger": {
+            "ref": [
+                "refs/heads/main",
+                "refs/heads/stable-*",
+                "refs/tags/**",
+                "refs/pull/**",
+            ],
+        },
+        "volumes": [
+            {
+                "name": "apps",
+                "temp": {},
+            },
+        ],
+    }]
+
+volumes = [
+    {
+        "name": "apps",
+        "path": "/apps",
+    },
+]
```

Comment:
this is not used but you can use it in above steps.

Reply:
Are you sure? I did what the docs says https://docs.drone.io/pipeline/kubernetes/syntax/volumes/host/

Reply:
ahh, now I got what you mean. fixed 👍